tests/unit-tests/internal/emails/test-class-email-customization.php

```diff
@@ -21,4 +21,44 @@ public function testInstance_WhenCalled_ReturnsInstance() {
 		/* Assert. */
 		$this->assertInstanceOf( Email_Customization::class, $result );
 	}
+
+	public function testInstace_WhenInitiated_AddsHookForRemovingLegacyEmail() {
+		/* Arrange. */
+		$settings = $this->createMock( Sensei_Settings::class );
+		$instance = Email_Customization::instance( $settings );
+
+		/* Act. */
+		$instance->init();
+
+		/* Assert. */
+		$this->assertEquals( 10, has_action( 'init', [ $instance, 'disable_legacy_emails' ] ) );
+	}
+
+	/**
+	 * Tests that the legacy email hooks are removed when the disable_legacy_emails function is called.
+	 *
+	 * @param string $action_name The name of the action.
+	 * @param string $function_name The name of the function.
+	 *
+	 * @dataProvider legacyHooksDataProvider
+	 */
+	public function testDisableLegacy_WhenCalled_RemovesLegacyEmailHooks( $action_name, $function_name ) {
+		/* Arrange. */
+		$settings        = $this->createMock( Sensei_Settings::class );
+		$instance        = Email_Customization::instance( $settings );
+		$priority_before = has_action( $action_name, [ \Sensei()->emails, $function_name ] );
+
+		/* Act. */
+		$instance->disable_legacy_emails();
+
+		/* Assert. */
+		$priority_after = has_action( $action_name, [ \Sensei()->emails, $function_name ] );
+		$this->assertNotEquals( $priority_before, $priority_after );
```

Comment:
```suggestion
		$this->assertFalse( $priority_after );
```
I think we want to check that we "de-registered" the function, don't we?

Reply:
I think this only allows us to ensure that the function is 'de-registered' at the end, but we don't know that it was or wasn't registered before. That's why I didn't use assertFalse